Fix example power button script, thanks to Timo Gurr.
http://bugs.gentoo.org/253764, Comment #11.

--- samples/powerbtn/powerbtn.sh
+++ samples/powerbtn/powerbtn.sh
@@ -24,6 +24,15 @@
         export XUSER=$user
 }
 
+getXconsole() {
+        console=`fgconsole`;
+        displaynum=`ps t tty$console | sed -n -re 's,.*/X .*:([0-9]+).*,\1,p'`
+        if [ x"$displaynum" != x"" ]; then
+                export DISPLAY=":$displaynum"
+                getXuser
+        fi
+}
+
 # Skip if we are just in the middle of resuming.
 test -f /var/lock/acpisleep && exit 0
 
